src/values/instruction_value.rs

```diff
@@ -684,14 +684,21 @@ impl<'ctx> InstructionValue<'ctx> {
         }
 
         let is_basic_block = unsafe { !LLVMIsABasicBlock(operand).is_null() };
-
         if is_basic_block {
             let bb = unsafe { BasicBlock::new(LLVMValueAsBasicBlock(operand)) };
 
-            Some(Operand::Block(bb.expect("BasicBlock should always be valid")))
-        } else {
-            Some(Operand::Value(unsafe { BasicValueEnum::new(operand) }))
+            return Some(Operand::Block(bb.expect("BasicBlock should always be valid")));
+        }
+
+        if unsafe { LLVMGetTypeKind(LLVMTypeOf(operand)) == LLVMTypeKind::LLVMMetadataTypeKind } {
+            // We may be dealing with null metadata, which would break memory invariants.
+            if LLVMIsAValueAsMetadata(operand).is_null() {
+                return None;
+            }
+            return Some(Operand::Metadata(MetadataValue::new(operand)));
         }
+
+        Some(Operand::Value(unsafe { BasicValueEnum::new(operand) }))
```

Comment:
This function is a little hard to follow the way that it's written. Maybe it should be an if/else chain instead?

Reply:
I've had to rewrite it a bit to make it compatible with all LLVM versions. I believe that the new version should be a bit easier to read.

I'm not a big fan of if/else chains, but if you think it's better, I can convert the code.